src/embedded.test.js

```diff
@@ -0,0 +1,608 @@
+import HelloSign from './embedded';
+import defaults from './defaults';
+import pkg from '../package.json';
+import settings from './settings';
+
+const mockSignURL = 'https://app.hellosign.com/editor/embeddedSign?signature_id=abcdef0123456789abcdef0123456789';
+const mockRequestURL = 'https://app.hellosign.com/editor/embeddedRequest?signature_id=abcdef0123456789abcdef0123456789';
+const mockClientId = 'abcdef0123456789abcdef0123456789';
+
+describe('HelloSign', () => {
+
+  afterEach(() => {
+    // Clear DOM.
+    document.getElementsByTagName('html')[0].innerHTML = '';
+  });
+
+  test('is defined', () => {
+    expect(HelloSign).toBeDefined();
+  });
+
+  test('instantiates without errors', () => {
+    expect(() => new HelloSign()).not.toThrow();
+  });
+
+  test('throws when instantiated without an object', () => {
+    expect(() => new HelloSign(false)).toThrow(/must be an object/);
+  });
+
+  describe('accessors', () => {
+
+    describe('element()', () => {
+      test('returns the base element', () => {
+        const client = new HelloSign({ clientId: mockClientId });
+
+        client.open(mockSignURL);
+
+        expect(client.element).toBeInstanceOf(HTMLElement);
+
+        client.close();
+      });
+    });
+
+    describe('isOpen()', () => {
+      test('returns the open state', () => {
+        const client = new HelloSign({ clientId: mockClientId });
+
+        expect(client.isOpen).toEqual(false);
+
+        client.open(mockSignURL);
+
+        expect(client.isOpen).toEqual(true);
+
+        client.close();
+
+        expect(client.isOpen).toEqual(false);
+      });
+    });
+  });
+
+  describe('methods', () => {
+
+    describe('open()', () => {
+
+      test('closes the old window if embedded is already open', (done) => {
+        const client = new HelloSign({
+          clientId: mockClientId,
+        });
+
+        const fn = jest.fn(() => {
+          expect(fn).toBeCalledTimes(1);
+
+          client.close();
+
+          done();
+        });
+
+        client.once(HelloSign.events.CLOSE, fn);
+
+        client.open(mockSignURL);
+        client.open(mockSignURL);
```

Comment:
Can `done()` be called twice? How does Jest know when to stop waiting?

Reply:
Oh I see you only bound your event listener once. Then what's the point of using `client.open(mockSignURL)` twice?